specs/required-inputs/README.md

```diff
@@ -0,0 +1,106 @@
+
+## Links To Relevant Work Items and Reference Material
+
+- [Nimble issue](https://github.com/ni/nimble/issues/2100)
+- [AzDO Feature](https://ni.visualstudio.com/DevCentral/_workitems/edit/2732543)
+- [Figma design](https://www.figma.com/design/PO9mFOu5BCl8aJvFchEeuN/Nimble_Components?node-id=1295-47481)
+
+## Implementation / Design
+
+### API
+
+Our FAST base components provide built-in forms support via the [same API](https://www.w3schools.com/tags/att_input_required.asp) exposed by the native HTML `input`:
+
+- `required`: boolean attribute whose presence indicates that a value must be provided to submit
+
+For radio buttons/groups, only the button elements expose the `required` attribute. If any of the radio buttons in a group are marked required, the group is treated as required. It does not matter if the button with `required` is disabled, hidden, etc. It is effectively as if the attribute is on the group of buttons, rather than a specific one. This matches the behavior of the native `input` with `type="radio"`. 
+
+### Visuals
+
+We will update our templates so that setting `required` causes a red asterisk to be displayed after the label, as per the visual design. For radio buttons/groups, the asterisk is on the group label, not individual button labels. We have already forked some of the templates from FAST, but we will now need to fork the ones for `nimble-number-field`, `nimble-radio-group`, and `nimble-text-field`. 
+
+This will be implemented using a styled span containing the text "*". The span must be outside of the `label` element so that it is not included in the accessible name. We will wrap the two in a `div` so that we can force a horizontal layout (`flex-direction: row`). This structure will also ensure that whatever the wrapping/ellipsizing behavior of the label is, the asterisk will always appear at the far right:
+
+```html
+<div class="annotated-label">
+    <label part="label" class="label">
+        <slot></slot>
+    </label>
+    ${when(x => x.required, html`
+        <span class="required-indicator">*</span>
+    `)}
+</div>
+```
+
+Note that for text-field, text-area, and number-field, we will also remove the visual if the control has the `readonly` attribute set, since [validation is not enforced on a readonly input](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly#attribute_interactions). This doesn't apply to select, combobox, or radio buttons, because they do not support `readonly`.
+
+Styles will be shared.
+
+### Form validation
+
+#### Vanilla HTML
+
+When the user attempts to submit a form and a `required` input is missing a value, a browser-specific indicator reports the validation error:
+
+![Missing value indicator](missing-value.png)
+
+We would expect this to already work, since the FAST components we derive from provide forms support. Unfortunately, this support is broken or incomplete for multiple components:
```

Comment:
It would be nice to break these out into issues for tracking in-case someone does rely on native form validation. We have at least one known user ([ozone for web login](https://dev.azure.com/ni/DevCentral/_git/Skyline?path=/Web/Workspaces/Ozone/WebLogin/src/app/app.component.html&version=GBmaster&line=8&lineEnd=32&lineStartColumn=1&lineEndColumn=16&lineStyle=plain&_a=contents) leveraging native form with `<nimble-text-field>` and `<nimble-button type="submit">`).

Reply:
Created bugs #2415, #2416, and #2417

I did not create bugs for the accessibility issues for the combobox and select, since they are within scope for this feature.